src/extension.ts

```diff
@@ -115,11 +166,32 @@ export function activate(): void {
         }
         sourcegraph.internal.updateContext(context)
     }
-    sourcegraph.configuration.subscribe(() => updateContext())
-    sourcegraph.workspace.onDidOpenTextDocument.subscribe(() => updateContext())
-    if (sourcegraph.workspace.onDidChangeRoots) {
-        sourcegraph.workspace.onDidChangeRoots.subscribe(() => updateContext())
-    }
+
+    // Update the context when the configuration, workspace roots or active editors change.
+    context.subscriptions.add(
+        combineLatest([
+            configurationChanges.pipe(
+                map(settings => settings['codecov.endpoints'])
+            ),
+            // Backcompat: rely on onDidChangeRoots if the extension host doesn't support rootChanges.
+            from(
+                sourcegraph.workspace.rootChanges
+                    ? sourcegraph.workspace.roots
```

Comment:
I think you meant this, given you are mapping it below `roots`.
```suggestion
                    ? sourcegraph.workspace.rootChanges
```
which also means you could replace the ternary with `||`